src/input_builders.jl

```diff
@@ -0,0 +1,99 @@
+"""
+    gustiness_constant(val)
+
+Create a constant gustiness parameterization with the given value.
+
+Returns a `ConstantGustinessSpec(val)` that provides a fixed gustiness 
+contribution to the effective windspeed.
+"""
+gustiness_constant(val) = ConstantGustinessSpec(val)
+
+# Velocity normalization helper (moved from types.jl)
+@inline function _normalize_velocity(u::Tuple{Any, Any}, ::Type{FT}) where {FT}
+    return (convert(FT, u[1]), convert(FT, u[2]))
+end
+function _normalize_velocity(u::AbstractVector, ::Type{FT}) where {FT}
+    length(u) == 2 ||
+        throw(ArgumentError("Velocity vectors must have two horizontal components."))
+    return (convert(FT, u[1]), convert(FT, u[2]))
+end
+_normalize_velocity(u::Nothing, ::Type{FT}) where {FT} = (zero(FT), zero(FT))
+_normalize_velocity(u::Tuple{}, ::Type{FT}) where {FT} = (zero(FT), zero(FT))
+
+
+"""
+    build_surface_flux_inputs(param_set, args...)
+
+Centralized helper that normalizes user-facing specifications (winds,
+roughness, gustiness, flux constraints) into a concrete `SurfaceFluxInputs`
+instance.
+"""
+function build_surface_flux_inputs(
+    param_set::APS,
+    T_int,
+    q_tot_int,
+    q_liq_int,
+    q_ice_int,
+    ρ_int,
+    T_sfc_guess,
+    q_vap_sfc_guess,
+    Φ_sfc,
+    Δz,
+    d,
+    u_int,
+    u_sfc,
+    config::SurfaceFluxConfig,
+    roughness_inputs,
+    flux_specs,
+    update_T_sfc,
+    update_q_vap_sfc,
+)
+    FT = promote_type(
```

Comment:
Do we need this? Could we just get FT from one of the input?

Reply:
It shouldn't be needed, but I see no harm: https://docs.julialang.org/en/v1/manual/conversion-and-promotion/